src/ServiceControl.UnitTests/Archiving/ArchiveGroupTests.cs

```diff
@@ -0,0 +1,68 @@
+﻿using NUnit.Framework;
+using Raven.Client;
+using ServiceControl.Contracts.Operations;
+using ServiceControl.MessageFailures;
+using ServiceControl.Recoverability;
+using ServiceControl.UnitTests.Recoverability;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ServiceControl.UnitTests.Archiving
+{
+    [TestFixture]
+    public class ArchiveGroupTests
+    {
+        [Test]
+        public void ArchiveGroup_skips_over_empty_batches_but_still_completes()
+        {
+            // Arrange
+            using (var documentStore = InMemoryStoreBuilder.GetInMemoryStore())
+            {
+                var groupId = "TestGroup";
+
+                using (var session = documentStore.OpenSession())
+                {
+                    var secondAchiveBatch = new ArchiveBatch { Id = ArchiveBatch.MakeId(groupId, ArchiveType.FailureGroup, 1) };
+                    session.Store(secondAchiveBatch);
+
+                    var archiveOperation = new ArchiveOperation
```

Comment:
Why is this needed? It feels like this test knows a lot about the inner workings of the archive subsystem.

Reply:
It does, it's specifically a unit test intended to test a specific scenario. In order to set that scenario up it needs to know what the data looks like.